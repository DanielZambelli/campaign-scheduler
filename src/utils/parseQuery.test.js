const {parseQuery} = require('./parseQuery')

describe(parseQuery, () => {

  it('validates', () => {
    const res = () => parseQuery({ lastName: { op: 'likexxx', value: 'doe' }})
    expect(res).toThrow('unsupported operator')
  })

  it('returns', () => {
    const res = parseQuery()
    expect(res).toMatchSnapshot()
  })

  it('returns', () => {
    const res = parseQuery({ name: undefined })
    expect(res).toMatchSnapshot()
  })

  it('returns', () => {
    const res = parseQuery({ name: false })
    expect(res).toMatchSnapshot()
  })

  it('returns', () => {
    const res = parseQuery({})
    expect(res).toMatchSnapshot()
  })

  it('returns', () => {
    const res = parseQuery({ name: 'john', lastName: 'doe' })
    expect(res).toMatchSnapshot()
  })

  it('returns', () => {
    const res = parseQuery({ limit: 1, order: [['id','ASC']], attributes: ['name'], name: 'john', lastName: { op: 'like', value: 'doe' }})
    expect(res).toMatchSnapshot()
  })

  it('returns', () => {
    const res = parseQuery({
      limit: 25,
      attributes: ['name','lastNane'],
      name: 'john',
      lastName: { op: 'like', value: 'doe' },
      age: { op: 'gte', value: 25 },
    })
    expect(res).toMatchSnapshot()
  })

})
