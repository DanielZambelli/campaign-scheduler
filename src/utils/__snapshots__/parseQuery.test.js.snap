// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseQuery returns 1`] = `
{
  "where": {},
}
`;

exports[`parseQuery returns 2`] = `
{
  "where": {},
}
`;

exports[`parseQuery returns 3`] = `
{
  "where": {
    "name": false,
  },
}
`;

exports[`parseQuery returns 4`] = `
{
  "where": {},
}
`;

exports[`parseQuery returns 5`] = `
{
  "where": {
    "lastName": "doe",
    "name": "john",
  },
}
`;

exports[`parseQuery returns 6`] = `
{
  "attributes": [
    "name",
  ],
  "limit": 1,
  "order": [
    [
      "id",
      "ASC",
    ],
  ],
  "where": {
    "lastName": {
      Symbol(like): "doe",
    },
    "name": "john",
  },
}
`;

exports[`parseQuery returns 7`] = `
{
  "attributes": [
    "name",
    "lastNane",
  ],
  "limit": 25,
  "where": {
    "age": {
      Symbol(gte): 25,
    },
    "lastName": {
      Symbol(like): "doe",
    },
    "name": "john",
  },
}
`;
